.github/workflows/publish-packages.yml

```diff
@@ -0,0 +1,89 @@
+name: Publish packages
+
+on:
+  push:
+    # possibly could be improved with a proper regex
+    # ex: https://stackoverflow.com/a/58869470
+    tags:
+      - v*.*.*
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+
+      - name: Install moreutils
+        run: brew install moreutils
+
+      - name: Install git-toolbelt
+        run:
+          brew install fzf && brew tap nvie/tap && brew install
+          nvie/tap/git-toolbelt
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.base_ref }}
+
+      - name: Set git tag
+        id: git_tag
+        run: echo "value=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
+
+      - name: Set release version
+        id: version
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: echo "value=${GIT_TAG#v}" >> $GITHUB_OUTPUT
+
+      - name: Set npm tag
+        id: npm_tag
+        env:
+          GIT_TAG: ${{ steps.git_tag.outputs.value }}
+        run: |
+          get_tag_if_exists () {
+            if grep -q "-" <<< "$1"; then
+                echo "${1##*-}" | sed 's/[0-9]//g'
+            else 
+                echo "latest"
+            fi
+          }
+          echo value=$(get_tag_if_exists "$GIT_TAG") >> $GITHUB_OUTPUT
+
+      - name: Check values
+        run: |
+          echo ${{steps.git_tag.outputs.value}}
+          echo ${{steps.version.outputs.value}}
+          echo ${{steps.npm_tag.outputs.value}}
+
+      - name: Setup git config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3.6.0
+        with:
+          node-version: 16.17.1
+
+      - name: Preparing release
+        env:
+          VERSION: ${{steps.version.outputs.value}}
+        run: ./scripts/release.sh -V "$VERSION"
+
+      - name: Build all packages
+        run: npm run build
+
+      # Add approve step here?
+
+      - name: Publish packages
+        # setup-node created .npmrc file to publish to npm
+        # https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+          VERSION: ${{ steps.version.outputs.value }}
+          TAG: ${{ steps.npm_tag.outputs.value }}
+        run: ./scripts/publish.sh -V "$VERSION" -t "$TAG"
```

Comment:
I'd love to move the `release.sh` and `publish.sh` scripts into the workflows directory, to keep them local here and to avoid them from being run by users locally.

Reply:
absolutely!